test/e2e/apimachinery/watchlist.go

```diff
@@ -123,3 +119,25 @@ func newSecret(name string) *v1.Secret {
 		ObjectMeta: metav1.ObjectMeta{Name: name},
 	}
 }
+
+type enabledWatchListClientFeatureGateRegistry struct {
+	originalGates clientfeatures.Gates
+}
+
+func newEnabledWatchListClientFeatureGateRegistry(originalGates clientfeatures.Gates) *enabledWatchListClientFeatureGateRegistry {
+	return &enabledWatchListClientFeatureGateRegistry{originalGates: originalGates}
+}
+
+func (r *enabledWatchListClientFeatureGateRegistry) Add(in map[clientfeatures.Feature]clientfeatures.FeatureSpec) error {
+	if _, ok := in[clientfeatures.WatchListClient]; !ok {
+		return fmt.Errorf("expected to find %q feature in the registry", clientfeatures.WatchListClient)
+	}
+	return nil
+}
+
+func (r *enabledWatchListClientFeatureGateRegistry) Enabled(feature clientfeatures.Feature) bool {
+	if feature == clientfeatures.WatchListClient {
+		return true
```

Comment:
Given that you force the gate here, why do we  need to wait for an env var change?  It appears the test used to force the feature to "on" and that after this change it is still force "on" regardless of the state of the env var being specified.

Reply:
sorry, I linked a wrong PR. We need https://github.com/kubernetes/kubernetes/pull/122571 (already merged).